### PR TITLE
Add integration test that covers STS flow from Envoy to Auth backend

### PR DESCRIPTION
**Pull request by JimmyCYJ**
Added a test framework which starts a STS server, a auth backend, a XDS server and an Envoy.
Envoy loads bootstrap config, and the config specifies gRPC connection to XDS server. The gRPC connection config specifies secure token exchange (STS) for dynamic token.
The gRPC library has a client that sends a STS request to the STS server. STS server calls auth backend for dynamic token, and STS server returns the dyanmic token to the client.
On receiving the dyanmic token, Envoy sets up XDS stream to the XDS server, and the server validates that dynamic token. After token validation, the XDS server starts pushing config over that stream.
A basic test is added using this framework.

https://github.com/istio/istio/issues/20133

### CODE REVIEW

File: security/pkg/stsservice/test/testdata/server-key.key [hunk 1]
```diff
@@ -0,0 +1,27 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (howardjohn) [inline security/pkg/stsservice/test/testdata/server-key.key]: where does this come from? WE have lots of issues of certs expiring and have no indication of how to regen

File: security/pkg/stsservice/mock/xdsserver.go [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+				c.lastReceivedToken = h[0]
+			}
+			if c.expectedToken != "" && strings.TrimPrefix(h[0], "Bearer ") != c.expectedToken {
+				c.t.Errorf("xDS stream (id: %d, url: %s) sent a token that does "+
+					"not match expected token (%s vs %s)", id, url, h[0], c.expectedToken)
+			} else {
+				c.t.Logf("xDS stream (id: %d, url: %s) has valid token: %v", id, url, h[0])
+			}
+		} else {
+			c.t.Errorf("XDS stream (id: %d, url: %s) does not have token in metadata %+v",
```
Comment (howardjohn) [inline security/pkg/stsservice/mock/xdsserver.go]: I don't think you can fail a test from a goroutine

Comment (JimmyCYJ) [reply security/pkg/stsservice/mock/xdsserver.go]: I think this works. I insert a c.t.Errorf("dummy error") before ```return nil``` here and the test fails with that error.
From https://golang.org/pkg/testing/#T, it says 
```A test ends when its Test function returns or calls any of the methods FailNow, Fatal, Fatalf, SkipNow, Skip, or Skipf. Those methods, as well as the Parallel method, must be called only from the goroutine running the Test function. The other reporting methods, such as the variations of Log and Error, may be called simultaneously from multiple goroutines.```

**Comment by howardjohn** [comment]
Ah just fatal has that issue. Thanks for the reference

File: security/pkg/stsservice/test/testdata/server-key.key (hunk 1 above)
Comment (JimmyCYJ) [reply security/pkg/stsservice/test/testdata/server-key.key]: These certs will expire in 2030, and they are from https://github.com/istio/istio/blob/master/tests/integration/security/sds_ingress/util/test_certs.go.
I generate them following this guide https://istio.io/docs/tasks/traffic-management/ingress/secure-ingress-mount/#generate-server-certificate-and-private-key

File: security/pkg/stsservice/test/testdata/trustworthy-jwt.jwt [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+jdHMvaGVsbG8taXN0aW8tMjM3MDIwL2xvY2F0aW9ucy91cy1jZW50cmFsMS1hL2NsdXN0ZXJzL2FkdmVudHVyZSJdLCJleHAiOjE
+1NjMzNDE5MDIsImlhdCI6MTU2MzI5ODcwMiwiaXNzIjoiaHR0cHM6Ly9jb250YWluZXIuZ29vZ2xlYXBpcy5jb20vdjEvcHJvamV
+jdHMvaGVsbG8taXN0aW8tMjM3MDIwL2xvY2F0aW9ucy91cy1jZW50cmFsMS1hL2NsdXN0ZXJzL2FkdmVudHVyZSIsImt1YmVybmV
+0ZXMuaW8iOnsibmFtZXNwYWNlIjoiZGVmYXVsdCIsInBvZCI6eyJuYW1lIjoic2xlZXAtNjk1YzY3Y2I1LTRxOTg1IiwidWlkIjo
+iODFhODdiNWItYTdmMC0xMWU5LWE5YjktNDIwMTBhODAwMWYxIn0sInNlcnZpY2VhY2NvdW50Ijp7Im5hbWUiOiJzbGVlcCIsInV
+pZCI6IjFmMmRkYjBmLTlkMTUtMTFlOS04NzViLTQyMDEwYTgwMDA4YiJ9fSwibmJmIjoxNTYzMjk4NzAyLCJzdWIiOiJzeXN0ZW0
+6c2VydmljZWFjY291bnQ6ZGVmYXVsdDpzbGVlcCJ9.qPpJN7p_G5yYNWgqLdWWivhPg2_ODr4dp6AR-uAPuW1a598jSH4r3Y4ZNy
+DShe4lepBamsmDdAL3cJnX-XfR38npYD6h6a-b2OU-ujWakSKrmN_fXha-EyHGmZj3u-fo-KxgR5OsiUWKDAHkmpmVevtAgA0USG
+gBOAZRK-biv1Gtw3ufgVcrlqT9A9dPuo1RE3h8_jFnfr8Gt3nIQ67jE9zwatggylMMUDeoQKHwF7ksCkAtEZmPj2JuOtuSYIgMuc
+GRGRJbd7qo8OngJDPCRrJqRV-7m-xf41GZoNnJEKvM8HbPwfW0htLAXFQr_IDPOAOoZ2sViJMi7Ww1HUfIew
```
[the remark this answers is not in the document]
Comment (JimmyCYJ) [reply security/pkg/stsservice/test/testdata/trustworthy-jwt.jwt]: I have replaced this with a random string, thanks.

File: security/pkg/stsservice/test/testdata/server-key.key (hunk 1 above)
[the remark this answers is not in the document]
Comment (JimmyCYJ) [reply security/pkg/stsservice/test/testdata/server-key.key]: We could set this dynamically. That would require some changes to the test framework. I would like to put it into a separate PR.

**Comment by JimmyCYJ** [comment]
Thanks a lot for the review! @liminw and @mandarjog

**Pull request merged**